File: práctica 8/ejercicio 1.py
class A:
    def __init__(self, x, z):
        self.x = x
        self.z = z

    def incrementaXZ(self):
        self.x += 1
        self.z += 1

    def incrementaZ(self):
        self.z += 1

class B:
    def __init__(self, y):
        self.y = y

    def incrementaYZ(self):
        self.y += 1
        self.z += 1 

    def incrementaZ(self):
        self.z += 1

class D(A, B):
    def __init__(self, x, y, z):
        A.__init__(self, x, z)
        B.__init__(self, y)

    def incrementaXYZ(self):
        self.x += 1
        self.y += 1
        self.z += 1

d = D(1, 2, 3)
d.incrementaXYZ()
print(d.x, d.y, d.z)

File: práctica 8/ejercicio 1.java
class A {
    int x, z;

    A(int x, int z) {
        this.x = x;
        this.z = z;
    }

    void incrementaXZ() {
        x++;
        z++;
    }

    void incrementaZ() {
        z++;
    }
}

class B {
    int y;
    A a;

    B(int y, A a) {
        this.y = y;
        this.a = a;
    }

    void incrementaYZ() {
        y++;
        a.z++;
    }

    void incrementaZ() {
        a.z++;
    }
}

class D extends A {
    B b;

    D(int x, int y, int z) {
        super(x, z);
        b = new B(y, this);
    }

    void incrementaXYZ() {
        x++;
        b.y++;
        z++;
    }

    void mostrar() {
        System.out.println(x + " " + b.y + " " + z);
    }
}

public class Main {
    public static void main(String[] args) {
        D d = new D(1, 2, 3);
        d.incrementaXYZ();
        d.mostrar();  
    }
}